src/helpers.js

```diff
@@ -0,0 +1,327 @@
+	const parentDir = dirname(currentDir);
+	return findProjectRoot(parentDir);
+};
+
+/**
+ * Gets the output path for the project.
+ *
+ * @param {string} dirname Directory name.
+ */
+export const getProjectOutputPath = (dirname) => {
+	const rootDir = findProjectRoot(process.cwd());
+
+	if (!rootDir) {
+		error('Unable to find project root.');
+		process.exit(1);
+	}
+
+	return join(rootDir, 'wp-content', dirname);
+};
+
+/**
+ * Empty the console.
+ */
+export const clearConsole = () => {
+	process.stdout.write('\x1Bc');
+};
+
+/**
+ * Writes the intro message to the console.
+ */
+export const writeIntro = () => {
+	const figletOpts = {
+		font: 'ANSI Regular',
+		width: 74,
+		whitespaceBreak: true,
+		verticalLayout: 'fitted',
+	};
+
+	const topBar = chalk.dim(`╭${'─'.repeat(76)}╮`);
+	const bottomBar = chalk.dim(`╰${'─'.repeat(76)}╯`);
+	const emptyBar = chalk.dim(`│${' '.repeat(76)}│`);
+	const midBar = chalk.dim(`├${'─'.repeat(76)}┤`);
+
+	// eslint-disable-next-line max-len
+	const processFiglet = (input) => input.split("\n").filter((line) => line.trim().length > 0).map((line) => `${chalk.dim('│')}  ${line.trim().padEnd(74, ' ')}${chalk.dim('│')}`).join("\n");
+	const processRedFiglet = (input) => input.map((line) => `${chalk.dim('│')}  ${chalk.redBright(line.padEnd(74, ' '))}${chalk.dim('│')}`).join("\n");
+	const processLine = (input) => `${chalk.dim('│')}  ${input.trim().padEnd(74, ' ')}${chalk.dim('│')}`;
+
+	const infinumLogoRaw = ["  ███████  ███████", "██       ██       ██", "██       ██       ██", "  ███████  ███████"];
+
+	const infinumLogo = processRedFiglet(infinumLogoRaw);
+	const esText = processFiglet(figlet.textSync('Eightshift', figletOpts));
+	const devKitText = processFiglet(figlet.textSync('DevKit', figletOpts));
+
+	// eslint-disable-next-line max-len
+	console.log([topBar, emptyBar, infinumLogo, emptyBar, esText, emptyBar, devKitText, emptyBar, midBar, emptyBar, processLine('Thank you for using Eightshift DevKit!'), emptyBar, bottomBar, ''].join("\n"));
+};
+
+/**
+ * Checks if the setup already exists.
+ *
+ * @param {string} folderPath Path to the folder.
+ * @param {string} type Type of the setup.
+ */
+export const checkIsSetupExists = (folderPath, type) => {
+	if (existsSync(folderPath)) {
+		console.log('');
+		error(`It looks like your all-ready have a setup ${type} in your project on path ${folderPath}. Please remove it and try again.`);
+		process.exit(1);
+	}
+}
+
+/**
+ * Install step.
+ *
+ * @param {string} describe Description of the step.
+ * @param {Promise} thisHappens Promise that needs to be resolved.
+ * @param {boolean} isFatal Is the step fatal.
+ */
+export const installStep = async ({ describe, thisHappens, isFatal = true }) => {
+	const spinner = ora({ text: describe, color: 'blue' }).start();
+
+	if (!thisHappens) {
+		throw new Error(`Missing 'thisHappens' parameter for step ${describe}, don't know what needs to be done at this step, aborting.`);
+	}
+
+	await thisHappens.then(() => {
```

Comment:
Either use `await` or `.then()`, but not both 😄

Reply:
heheh taken from the previous version :D I hope on you to fix this 🗡️

Reply:
```js
try {
	await thisHappens();
	spinner.succeed();
} catch (exception) {
	console.log(exception);
	spinner.fail();
	error(exception);

	if (isFatal) {
		error(`'${describe}' was a required step, exiting.`);
		process.exit(1);
	}
}
```